### PR TITLE
#19 OCC

### PR DESCRIPTION
**Pull request by VsSekorin**
As per #19 
`OCC.java` introduced

Now the class contains a **very dirty code**. I would like to receive advice on correcting.

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256 please, pay attention to this pull request

**Comment by 0crat** [comment]
Job `gh:yegor256/jpeek#21` is in scope.

File: src/main/java/org/jpeek/metrics/cohesion/OCC.java [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+import org.jpeek.metrics.Colors;
+import org.jpeek.metrics.JavassistClasses;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.xembly.Directive;
+
+/**
+ * Optimistic Class Cohesion (OCC).
```
Comment (yegor256) [inline src/main/java/org/jpeek/metrics/cohesion/OCC.java]: @VsSekorin please, add a brief explanation of the logic here, like it's done in `LCOM.java`

File: src/main/java/org/jpeek/metrics/cohesion/OCC.java [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Vseslav Sekorin (vssekorin@gmail.com)
+ * @version $Id$
+ * @see <a href="https://www.researchgate.net/publication/268046583_A_Proposal_of_Class_Cohesion_Metrics_Using_Sizes_of_Cohesive_Parts">A Proposal of Class Cohesion Metrics Using Sizes of Cohesive Parts</a>
+ * @since 0.2
```
Comment (yegor256) [inline src/main/java/org/jpeek/metrics/cohesion/OCC.java]: @VsSekorin should be 0.4

File: src/test/java/org/jpeek/metrics/cohesion/OCCTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        );
+    }
+
+    @Test
+    public void createsXmlReportForYellowClass() throws IOException {
+        MatcherAssert.assertThat(
+            XhtmlMatchers.xhtml(
+                new Xembler(
+                    new OCC(
+                        new FakeBase("Test")
```
Comment (yegor256) [inline src/test/java/org/jpeek/metrics/cohesion/OCCTest.java]: @VsSekorin can you give this class a more meaningful name? You're not using `Foo` for some reason. What is that reason? The name should say it. Make it `TooComplex` or `TwoAttributes`, something like that, to explain us what these classes are about.

File: src/test/java/org/jpeek/metrics/cohesion/OCCTest.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        );
+    }
+
+    @Test
+    public void createsXmlReportForRedClass() throws IOException {
+        MatcherAssert.assertThat(
+            XhtmlMatchers.xhtml(
+                new Xembler(
+                    new OCC(
+                        new FakeBase("C")
```
Comment (yegor256) [inline src/test/java/org/jpeek/metrics/cohesion/OCCTest.java]: @VsSekorin the same here, `C` is not really a good name :)

**Comment by yegor256** [comment]
@VsSekorin indeed, it looks complex, but not messy. You can move that inner class out of the method, but it seems that such a refactoring will only decrease readability. I would leave everything the way it is now and only fix the things I mentioned above.

**Comment by yegor256** [comment]
@VsSekorin good code, by the way, thanks!

**Comment by yegor256** [comment]
@VsSekorin one more thing, don't forget to update the README and explain the metric there.

**Comment by yegor256** [comment]
@VsSekorin and please add the PDF to this folder: https://github.com/yegor256/jpeek/tree/master/papers

**Comment by VsSekorin** [comment]
@yegor256 thanks for your review.

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**